scipy/optimize/optimize.py

```diff
@@ -953,6 +954,22 @@ def check_grad(func, grad, x0, *args, **kwargs):
     epsilon : float, optional
         Step size used for the finite difference approximation. It defaults to
         ``sqrt(np.finfo(float).eps)``, which is approximately 1.49e-08.
+    random_projection : bool, optional
+        If set to ``True``, then gradients along a random vector 
+        are used to check `grad` against forward difference approximation 
+        using `func`. By default it is ``False``.
+    seed : {None, int, `numpy.random.Generator`,
+            `numpy.random.RandomState`}, optional
+
```

Comment:
This should clarify up front that it's only used if `random_projection=True` (or `direction='random'`, if we make that change)

Reply:
Yeah. I removed that part in a commit. I will add that back. Thanks for pointing out.

Reply:
This is also done.